src/Integration.Vsix.UnitTests/CFamilyTest.cs

```diff
@@ -0,0 +1,262 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Integration.Vsix;
+using System.Collections.Generic;
+using FluentAssertions;
+using System;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests
+{
+    [TestClass]
+    public class CFamilyTest
+    {
+        private const string FileName = @"C:\absolute\path\to\file.cpp";
+        [TestMethod]
+        public void FileConfig_Test1()
+        {
+            string sqLanguage;
+            CFamily.Capture[] captures = new CFamily.FileConfig()
+            {
+                PlatformName = "Win32",
+                PlatformToolset = "v140",
+                IncludeDirectories = "sys1;sys2;",
+                AdditionalIncludeDirectories = "dir1;dir2;",
+                PreprocessorDefinitions = "D1;D2;",
+                UndefinePreprocessorDefinitions = "U1;U2;",
+                ForcedIncludeFiles = "h1;h2;",
+                CompileAs = "Default",
+                CompileAsManaged = "",
+                RuntimeLibrary = "MultiThreaded",
+                ExceptionHandling = "false",
+                EnableEnhancedInstructionSet = "NotSet",
+                RuntimeTypeInfo = "true",
+                BasicRuntimeChecks = "Default",
+                AdditionalOptions = "/a1 /a2",
+                AbsoluteFilePath = FileName,
+            }.ToCaptures(FileName, out sqLanguage);
+            CFamily.Capture p = captures[0];
+            CFamily.Capture c = captures[1];
+
+            p.Compiler.Should().Be("msvc-cl");
+            p.StdErr.Should().Be("19.00.00 for x86");
+            p.Cwd.Should().Be(@"C:\absolute\path\to");
+            p.Executable.Should().Be("cl.exe");
+
+            c.Compiler.Should().Be("msvc-cl");
+            c.StdErr.Should().BeNull("otherwise will be considered as probe");
+            c.Cwd.Should().Be(p.Cwd);
+            c.Executable.Should().BeSameAs(p.Executable, "otherwise won't be associated with probe");
+            c.Env.Should().Equal(new List<string>()
```

Comment:
Just a tip - you could use `new [] { "..." }` instead, it is shorter, but still, nothing wrong with List<string>